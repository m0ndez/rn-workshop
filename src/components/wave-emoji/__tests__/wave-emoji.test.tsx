import React from "react";
import { render } from "@testing-library/react-native";
import { WaveEmoji } from "../wave-emoji";

describe("WaveEmoji component", () => {
  it("renders without crashing", () => {
    render(<WaveEmoji />);
  });

  it("renders the provided emoji", () => {
    const emoji = "ðŸ‘‹";
    const { getByText, toJSON } = render(<WaveEmoji emoji={emoji} />);
    const emojiText = getByText(emoji);
    expect(emojiText).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
