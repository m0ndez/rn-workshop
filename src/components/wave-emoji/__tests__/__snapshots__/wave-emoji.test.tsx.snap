// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaveEmoji component renders the provided emoji 1`] = `
<View
  collapsable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    {
      "transform": [
        {
          "rotate": "0deg",
        },
      ],
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontSize": 28,
          "lineHeight": 32,
          "marginTop": -6,
        },
      ]
    }
  >
    ðŸ‘‹
  </Text>
</View>
`;
